### PR TITLE
Core/SocialMgr: fix case when contacts appears in friend list many times

### LINKED ISSUE #19266: Core: Duplicated entry on friend list
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

**Description:** On the friend list some contacts appears many times.

**Current behaviour:** http://i.imgur.com/1TJ2zQ8.jpg
http://i.imgur.com/oV7igXl.jpg

**Expected behaviour:** The friend must appear only once

**Steps to reproduce the problem:**

Unknown
Nothing weird or unusual on the DB

**Branch(es):** 3.3.5

**TC rev. hash/commit:** rev. 8089f1816c9b

### PR DESCRIPTION
**Pull request by ForesterDev**
**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #19266  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.) tested

** How to reproduce bug: add contacts to ignore list.

### CODE REVIEW

File: src/server/game/Entities/Player/SocialMgr.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ void PlayerSocial::SendSocialList(Player* player, uint32 flags)
 {
     ASSERT(player);
 
-    uint32 size = _playerSocialMap.size();
+    uint32 size = GetNumberOfSocialsWithFlag(SocialFlag(flags));
```
Comment (jackpoz) [inline src/server/game/Entities/Player/SocialMgr.cpp]: is this the same as the count variable below ?

Comment (ForesterDev) [inline src/server/game/Entities/Player/SocialMgr.cpp]: Yes, but we need this variable to initialize SMSG_CONTACT_LIST packet size.

**Comment by ForesterDev** [comment]
@jackpoz i think this solution will be vore cleaner

**Comment by jackpoz** [comment]
It looks better imo. Could you repeat the tests just to be sure ? Then I think it can be merged.

**Comment by ForesterDev** [comment]
@jackpoz works fine. 

I also have a proposal: rename FriendInfo struct to ContactInfo, because not only friends, but also ignored players use this strcture

**Comment by jackpoz** [comment]
Better keep refactoring changes for another PR and focus only on the bug in this one

File: src/server/game/Entities/Player/SocialMgr.cpp [hunk 2]
```diff
@@ -125,40 +125,43 @@ void PlayerSocial::SendSocialList(Player* player, uint32 flags)
...
     {
-        if (!(v.second.Flags & flags))
+        FriendInfo friendInfo = v.second;
+        if (!(friendInfo.Flags & flags))
             continue;
 
-        ++count;
-        sSocialMgr->GetFriendInfo(player, v.first, v.second);
+        ObjectGuid friendGUID = v.first;
+        sSocialMgr->GetFriendInfo(player, friendGUID, friendInfo);
```
Comment (ariel-) [inline src/server/game/Entities/Player/SocialMgr.cpp]: You now broke functionality that kept map info updated, because you get the friendInfo in a copy.

**Pull request closed without merging**